test/lib/validation/schema/specVersion/2.0/kind/project/application.js

```diff
@@ -34,956 +35,661 @@ test.after.always((t) => {
+			},
+			{
+				dataPath: "/builder/bundles/1/bundleDefinition/sections/1/mode",
+				keyword: "enum",
+				message: "should be equal to one of the allowed values",
+				params: {
+					allowedValues: [
+						"raw",
+						"preload",
+						"require",
+						"provided",
+					],
+				},
+				schemaPath: "../project.json#/definitions/builder-bundles/items/properties/bundleDefinition/properties/sections/items/properties/mode/enum",
+			},
+			{
+				dataPath: "/builder/bundles/1/bundleDefinition/sections/1/filters",
+				keyword: "type",
+				message: "should be array",
+				params: {
+					type: "array",
+				},
+				schemaPath: "../project.json#/definitions/builder-bundles/items/properties/bundleDefinition/properties/sections/items/properties/filters/type",
+			},
+			{
+				dataPath: "/builder/bundles/1/bundleOptions",
+				keyword: "additionalProperties",
+				message: "should NOT have additional properties",
+				params: {
+					additionalProperty: "notAllowed",
+				},
+				schemaPath: "../project.json#/definitions/builder-bundles/items/properties/bundleOptions/additionalProperties",
+			},
+			{
+				dataPath: "/builder/bundles/1/bundleOptions/optimize",
+				keyword: "type",
+				message: "should be boolean",
+				params: {
+					type: "boolean",
+				},
+				schemaPath: "../project.json#/definitions/builder-bundles/items/properties/bundleOptions/properties/optimize/type",
+			},
+			{
+				dataPath: "/builder/bundles/1/bundleOptions/numberOfParts",
+				keyword: "type",
+				message: "should be number",
+				params: {
+					type: "number",
+				},
+				schemaPath: "../project.json#/definitions/builder-bundles/items/properties/bundleOptions/properties/numberOfParts/type",
+			},
+			{
+				dataPath: "/builder/componentPreload",
+				keyword: "additionalProperties",
+				message: "should NOT have additional properties",
+				params: {
+					additionalProperty: "path",
+				},
+				schemaPath: "../project.json#/definitions/builder-componentPreload/additionalProperties",
+			},
+			{
+				dataPath: "/builder/componentPreload/paths",
+				keyword: "type",
+				message: "should be array",
+				params: {
+					type: "array",
+				},
+				schemaPath: "../project.json#/definitions/builder-componentPreload/properties/paths/type",
+			},
+			{
+				dataPath: "/builder/componentPreload/namespaces",
+				keyword: "type",
+				message: "should be array",
+				params: {
+					type: "array",
+				},
+				schemaPath: "../project.json#/definitions/builder-componentPreload/properties/namespaces/type",
+			}
+		]);
 	});
-});
 
-test("metadata.deprecated: false", async (t) => {
-	await assertValidation(t, {
-		"specVersion": "2.0",
-		"type": "application",
-		"metadata": {
-			"name": "my-application",
-			"deprecated": false
-		}
-	});
-});
+	test("framework configuration: Invalid (specVersion " + version + ")", async (t) => {
```

Comment:
I think all framework tests should be common across application, library and theme-library.

Reply:
these are run for all kinds but not for all versions
(added comments)